src/x86/gpio.c

```diff
@@ -0,0 +1,105 @@
+#include "gpio.h"
+
+#include <stdbool.h>
+#include <stdint.h>
+
+// TODO(ckitagawa): This whole implementation works and is fast but is bad due to the need for
+// sourcing configurations. Figure out an alternative. Compared to registers it is memory
+// inefficient but it saves having to define fake registers and stores the same information.
+
+// TODO(ckitagawa): Move this to a configuration file which is included in the x86 shared header.
+#define MAX_PORTS 6
+#define MAX_PINS 16
+
+// TODO(ckitagawa): Move to a shared header to make this publically available to share between all
+// x86 HAL implementations. They can then read from these to get information on current settings.
+static GPIOSettings pin_settings[MAX_PORTS * MAX_PINS];
+uint8_t gpio_x86_pin_input_value[MAX_PORTS * MAX_PINS];
```

Comment:
Is there a reason why this isn't static?

Note that since we're in the x86, there isn't really any point in prefixing things with _x86.